docs/architecture/grpc-for-wcf-developers/load-balancing.md

```diff
@@ -1,11 +1,44 @@
 ---
-title: Load-balancing gRPC - gRPC for WCF Developers
-description: TO BE WRITTEN
+title: Load balancing gRPC - gRPC for WCF Developers
+description: Choosing a load balancer to work with gRPC services.
 author: markrendle
 ms.date: 09/02/2019
 ---
 
-# Load-balancing
+# Load balancing gRPC
+
+A typical deployment of a gRPC application includes a number of identical instances of the service, providing resilience and horizontal scalability. Load balancing distributed incoming requests across these instances to provide full usage of all available resources. To make this load balancing invisible to the client, it's common to use a proxy load balancer server to handle requests from clients and route them to back-end instances.
+
+Load balancers are classified according to the *layer* they operate on. Layer 4 load balancers work on the *transport* level, for example, with TCP sockets, connections and packets. Layer 7 load balancers work at the *application* level, specifically handling HTTP/2 requests for gRPC applications.
+
+## L4 load balancers
+
+An L4 load balancer accepts a TCP connection request from a client, opens another connection to one of the back-end instances, and copies data between the two connections with no real processing. L4 offers excellent performance and low latency, but very little control or intelligence. As long as the client keeps the connection open, all requests will be directed to the same back-end instance.
+
+An example of an L4 load balancer is the [Azure Load Balancer](https://azure.microsoft.com/services/load-balancer/).
+
+## L7 load balancers
+
+An L7 load balancer parses incoming HTTP/2 requests and passes them on to back-end instances on a request-by-request basis, no matter how long the connection is held by the client.
```

Comment:
Is there any difference in behavior when using streaming? Any limitations or anything needed to be accounted for when using an L7 LB? What about TLS termination, does the L7 LB need to terminate the TLS session or does the channel support in HTTP/2 avoid this?

Reply:
I'm not aware of any limitations in the L7 load balancers that work with gRPC. TLS is mentioned a couple of paragraphs down.